### PR TITLE
WFCORE-1561 restore truncated result address

### PR DESCRIPTION
**Pull request by luck3y**
This restores the ProxyAddressTranslated address for a result by providing a method in ProxyAddressTranslater to merge the original op address, and the result. Thoughts / comments appreciated, as always.

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/ProxyOperationAddressTranslator.java [hunk 1]
```diff
@@ -48,6 +55,31 @@ public PathAddress translateAddress(PathAddress address) {
                 return translated;
             }
 
+            public PathAddress restoreAddress(PathAddress opAddress, PathAddress translatedAddress) {
+                // original is /host=foo/server=bar/subsystem=*
+                // translated is /subsystem=*
+                // we want the result to be translated back to /host=foo/server=bar/subsystem=quux
```
Comment (kabir) [inline controller/src/main/java/org/jboss/as/controller/ProxyOperationAddressTranslator.java]: This comment stumped me a bit. In my head an address with wildcards should expand into a list. After reading the Jira it makes sense though.

Comment (luck3y) [inline controller/src/main/java/org/jboss/as/controller/ProxyOperationAddressTranslator.java]: I agree this isn't explicit enough, I'll expand on the comment to make it clearer.

**Comment by luck3y** [comment]
Yeah, actually that's a case I was a bit concerned about, where the same path key could occur > 1 times,    I'll add another test and fix the code.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
@bstansberry I pushed, and the fixme didn't drop. Anyway, how does this look? Test added and I retested against full and I get the expected full addresses for everything I can think of to test.

**Comment by bstansberry** [comment]
@luck3y Looks good. Our github integration seems to be broken; besides the fixme it doesn't appear to have rerun CI either. I'll do that manually as part of merge processing.

**Pull request merged**